### ISSUE TITLE
[0.9.4.2] - DERIVATIVE - Shows non-aggregated value

### ISSUE THREAD

**Issue by mkrsn** [influxdata/influxdb#4852]
Hi,

first of all: The influxdb is really great! I'm testing it since last week with collectd and grafana - I really love it!

Today i was building some graphes and found a "bug" in DERIVATIVE and NON_NEGATIVE_DERIVATIVE. In some situations the aggregator shows in the first row the original value instead of the aggregated one. This cause a big Peak in Grafana which screwed the Graph & Legend Values up.

Example
```
> select derivative(mean(value)) from apache_value where host='schnipp' and instance='dynamic' and type='apache_requests' and time > now() - 15m group by time(10s), host, instance, type fill(0) limit 5
name: apache_value
tags: host=schnipp, instance=dynamic, type=apache_requests
time			derivative
----			----------
2015-11-20T13:49:30Z	1.95241e+06
2015-11-20T13:49:40Z	23
2015-11-20T13:49:50Z	25
2015-11-20T13:50:00Z	23


> select derivative(mean(value)) from apache_value where host='schnipp' and instance='dynamic' and type='apache_requests' and time > now() - 15m group by time(10s), host, instance, type fill(0) limit 5
name: apache_value
tags: host=schnipp, instance=dynamic, type=apache_requests
time			derivative
----			----------
2015-11-20T13:49:30Z	23
2015-11-20T13:49:40Z	23
2015-11-20T13:49:50Z	25
2015-11-20T13:50:00Z	23
```

![bildschirmfoto 2015-11-20 um 14 56 03](https://cloud.githubusercontent.com/assets/10330231/11301390/0ad4212e-8f97-11e5-8bb4-8712ce166a5b.png)

Actually i workaround this Problem with "limit 999999 offset 1" which is not really great - but working at the moment ^^

Best regards from Germany

**Comment by MatthiasKrause** [comment]
had to edit my example because of a mistake.

**Comment by cnelissen** [comment]
I have run into this problem as well, it has to do with the fill(0) function...

For example, starting with a new dataset, insert a value (something > 0 because data collection on this metric has been running for some period before we decided to start collecting it)

```
time                                  value
2015-11-20T12:00:00Z    987466535764
2015-11-20T12:05:00Z    987466847565
```

Now we run a derivative function with fill(0):

```
SELECT DERIVATIVE(MEAN(value), 1s) FROM "measurement" GROUP BY time(5m) fill(0)

time                                 value
2015-11-20T12:00:00Z   3291555119.21333333333333
2015-11-20T12:05:00Z   1039.33666666666667
```

The fill(0) is inserting an empty value for the 11:55 slot, which means the derivative between 11:55 and 12:00 is the entire new value at that point, which is not correct.

I think the derivative function should automatically ignore the first value in a dataset because it really can't determine the rate between one unknown value and one known value, it can only accurately determine a rate between two known values, and any attempt to "fill" (either 0, or previous) it will only distort the data.

**Comment by jwestboston** [comment]
I'm running into this issue with InfluxDB 0.10.3.

It seems to only be an issue when the GROUP BY time(interval) has an interval different than the rate of series input.  In other words, if I insert data every 10s and GROUP BY time(10s) I dont see an issue.  But if I insert data every 10s and GROUP BY time(60s) the first value is always wayyy off.

And like you said, this presents a huge problem with graphing!

**Comment by jwestboston** [comment]
Changing to fill(none) or fill(null) also does not help.

**Comment by jwestboston** [comment]
This seems also to be only an issue when the aggregate function is SUM().

For example

SELECT DERIVATIVE(SUM("field"), 1s)

it is not the case when the aggregate function is MEAN() for me.  But it looks like this was the case for @cnelissen

**Comment by chrusty** [comment]
I've got the same thing happening here too. Derivative() & non-negative-derivative() often return a first-bucket row so large that it breaks the perspective on any graphs i'm generating. This is also using the SUM() aggregator.

**Comment by jwestboston** [comment]
@chrusty Yep.  Happening here too.  Very frustrating.

**Comment by jsternberg** [comment]
This is a duplicate of #3247 and will be fixed in 0.13. Please try out the version on master or in the nightly to confirm if this is fixed. If it isn't, please reopen with information about what you did so I can try to fix it before 0.13 is released.

Thanks!
